modules/lib/src/main/resources/assets/js/app/view/context/ContextSplitPanel.ts

```diff
@@ -2,134 +2,262 @@ import {ResponsiveManager} from 'lib-admin-ui/ui/responsive/ResponsiveManager';
+        const parentWidth: number = this.getParentElement().getEl().getWidthWithBorder();
+        const letPanelFloatingModeResponsiveRange: ResponsiveRange = this.getLeftPanelResponsiveRangeToSwitchToFloatingMode();
 
-        this.contextPanelsManager = nonMobileContextPanelsManagerBuilder.build();
+        // Calculate context panel with half width of the splitter, since context panel in floating mode
+        // is bigger on that value.
+        const contextPanelWidth: number = this.getActiveWidthPxOfSecondPanel();
+        const halfSplitter: number = this.getSplitterThickness() / 2;
+        const leftPanelExpectedWidth: number = parentWidth - (contextPanelWidth + halfSplitter);
+
+        return letPanelFloatingModeResponsiveRange.isFitOrSmaller(leftPanelExpectedWidth)
+               ? ContextPanelMode.FLOATING
+               : ContextPanelMode.DOCKED;
+    }
+
+    protected getLeftPanelResponsiveRangeToSwitchToFloatingMode(): ResponsiveRange {
+        return ResponsiveRanges._720_960;
+    }
+
+    private isContextPanelLessThanMin(): boolean {
+        return this.getActiveWidthPxOfSecondPanel() < ContextSplitPanel.CONTEXT_MIN_WIDTH;
     }
 
-    private isInsideWizard(): boolean {
-        return this.data != null;
+    hideContextPanel(): void {
+        this.foldSecondPanel();
+        this.setState(ContextPanelState.COLLAPSED);
     }
 
-    private isPageEditorPresent(): boolean {
-        return this.isInsideWizard() && this.data.liveFormPanel != null;
+    showContextPanel(): void {
+        this.switchPanelModeIfNeeded();
+
+        if (this.isContextPanelLessThanMin()) {
+            this.setActiveWidthPxOfSecondPanel(SplitPanelSize.Pixels(ContextSplitPanel.CONTEXT_MIN_WIDTH));
+        }
+
+        this.showSecondPanel();
+        this.setState(ContextPanelState.EXPANDED);
     }
 
-    private renderAfterDockedPanelReady() {
-        this.floatingContextPanel.insertAfterEl(this);
+    private doHandleResizeEvent(): void {
+        if (this.isCollapsed()) {
+            return;
+        }
+
+        this.switchPanelModeIfNeeded();
 
-        if (this.contextPanelsManager.requiresCollapsedContextPanel()) {
-            this.contextPanelsManager.hideDockedContextPanel();
+        if (this.isContextPanelLessThanMin()) {
+            this.setActiveWidthPxOfSecondPanel(SplitPanelSize.Pixels(ContextSplitPanel.CONTEXT_MIN_WIDTH + this.getSplitterThickness()));
+            this.distribute();
         }
+    }
+
+    private renderAfterDockedPanelReady(): void {
+        this.hideContextPanel();
+        this.subscribeContextPanelsOnEvents();
+    }
 
-        this.subscribeContextPanelsOnEvents(this.contextPanelsManager);
+    private requiresCollapsedContextPanel(): boolean {
+        const totalWidth: number = Body.get().getEl().getWidthWithBorder();
+        return this.getExpectedContextPanelMode() === ContextPanelMode.FLOATING ||
+               ResponsiveRanges._1620_1920.isFitOrSmaller(totalWidth);
```

Comment:
Swap the two conditions so that a simpler condition is called before the one with all the calculations.